### PR TITLE
Specify Python versions for some requirements

### LINKED ISSUE #842: Cherry pick requirement specifiers from PR 815 for maint-1.8
Obsoletes issue #841.

### PR DESCRIPTION
**Pull request by sgillies**
Resolves #842

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -279,19 +279,16 @@ def run(self):
...
-
-if sys.version_info < (2, 7):
-    requirements.append('argparse')
-    requirements.append('ordereddict')
-
-if sys.version_info < (3, 4):
-    requirements.append('enum34')
+    'munch',
+    'argparse; python_version <= "3.4"',
+    'ordereddict; python_version <= "2.7"',
```
Comment (rbuffat) [inline setup.py]: Should ordereddict be < 2.7 and not <= 2.7? https://pypi.org/project/ordereddict/
Python 2.7.0 was released July 3, 2010, ordered dict 1.1 Januar 14, 2010. 

There was previously a discrepancy between setup.py ( < 2.7) and requirements.txt (<= 2.7). Now both is <= 2.7.

**Comment by jorisvandenbossche** [comment]
Also the version specification for argparse changed (from "< 2.7" to "<= 3.4"), while argparse should already be availble on python 2.7 I think: https://docs.python.org/2/library/argparse.html

I am running into some problems related to argparse on the python 2.7 conda-forge builds that I don't fully understand: https://github.com/conda-forge/fiona-feedstock/pull/155

**Comment by jorisvandenbossche** [comment]
So I think the wrong argparse version specification makes the fio CLI fail, something similar as https://github.com/Gallopsled/pwntools/issues/598

**Comment by jorisvandenbossche** [comment]
Confirmed in the conda-forge build, we needed this patch to get the tests passing (https://github.com/conda-forge/fiona-feedstock/pull/155/files#diff-dc29257d4ab27f711c103e7cc6de40a4):

```patch
--- Fiona-1.8.12.orig/setup.py	2019-12-05 03:08:55.000000000 -0300
+++ Fiona-1.8.12/setup.py	2019-12-05 08:50:47.017441535 -0300
@@ -280,7 +280,7 @@
     'click-plugins>=1.0',
     'six>=1.7',
     'munch',
-    'argparse; python_version <= "3.4"',
+    'argparse; python_version <= "2.6"',
-     'ordereddict; python_version <= "2.7"',
+     'ordereddict; python_version <= "2.6"',
     'enum34; python_version < "3.4"'
 ]
```

**Pull request merged**